### ISSUE TITLE
ACR login fails with latest version of Azure CLI

### ISSUE THREAD

**Issue by poloagustin** [CircleCI-Public/azure-acr-orb#11]
### Orb version
0.1.3

### What happened

The latest version of the [azure-cli](https://docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?toc=%2Fcli%2Fazure%2Ftoc.json&bc=%2Fcli%2Fazure%2Fbreadcrumb%2Ftoc.json&view=azure-cli-latest#february-18-2020) introduced the following change:
`az acr login: Throw a CLIError if there are errors returned by docker command` 

This breaks the step **Log into ACR** with the following output:
```
#!/bin/bash -eo pipefail
az acr login --name $AZURE_REGISTRY_NAME_STAGING
Login Succeeded
docker: WARNING! Your password will be stored unencrypted in /home/circleci/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Exited with code exit status 1
```
### Expected behavior

The docker login command should exit successfully

**Comment by AustinFelipe** [comment]
@poloagustin, I'm facing this problem as well. Do you have any workaround?

**Comment by poloagustin** [comment]
@AustinFelipe 
Yeah, for the time being I created a custom job that installs a previous version of the cli
Here's a quick example:
```
executors:
  default: azure-acr/default
commands:
  my-build-and-push-image: azure-acr/build-and-push-image
  push-image:
    executor: default-executor
    steps:
      - run:
          command: |
            # Verify the CLI isn't already installed
            # Use which instead of command -v for wider coverage of envs

            if which az > /dev/null; then
              echo "Azure CLI installed already."
              exit 0
            fi

            # Set sudo to work whether logged in as root user or non-root user

            if [[ $EUID == 0 ]]; then export SUDO=""; else export SUDO="sudo"; fi

            $SUDO apt-get update && $SUDO apt-get -qqy install apt-transport-https

            if [[ $(command -v lsb_release) == "" ]]; then
              echo "Installing lsb_release"
              $SUDO apt-get -qqy install lsb-release
            fi

            # Create an environment variable for the correct distribution
            export AZ_REPO=$(lsb_release -cs)

            # Modify your sources list

            echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
              $SUDO tee /etc/apt/sources.list.d/azure-cli.list

            if [[ $(command -v curl) == "" ]]; then
              echo "Installing curl"
              $SUDO apt-get -qqy install curl
            fi

            # Get the Microsoft signing key

            curl -L https://packages.microsoft.com/keys/microsoft.asc | $SUDO apt-key add -

            # Update and install the Azure CLI

            $SUDO apt-get update
            $SUDO apt-get -qqy install \
              ca-certificates \
              azure-cli=2.0.81-1~xenial
            echo "Azure CLI is now installed."
          name: Install Azure CLI Version 2.0.81, if not available
      - my-build-and-push-image:
          attach-workspace: <<parameters.attach-workspace>>
          checkout: <<parameters.checkout>>
          login-server-name: <<parameters.login-server-name>>
          registry-name: <<parameters.registry-name>>
          repo: <<parameters.repo>>
          tag: <<parameters.tag>>
          path: <<parameters.path>>
    parameters:
      alternate-tenant:
        default: false
        description: Use an alternate tenant. Only applicable for user logins.
        type: boolean
      attach-workspace:
        default: false
        description: |
          Boolean for whether or not to attach to an existing workspace. Default is false.
        type: boolean
      azure-password:
        default: AZURE_PASSWORD
        description: |
          Environment variable storing your Azure password. Only applicable for
          user logins.
        type: env_var_name
      azure-sp:
        default: AZURE_SP
        description: |
          Name of environment variable storing the full name of the Service
          Principal, in the form http://app-url. Only applicable for Service
          Principal logins.
        type: env_var_name
      azure-sp-password:
        default: AZURE_SP_PASSWORD
        description: |
          Name of environment variable storing the password for the Service
          Principal. Only applicable for Service Principal logins.
        type: env_var_name
      azure-sp-tenant:
        default: AZURE_SP_TENANT
        description: |
          Name of environment variable storing the tenant ID for the Service Principal. Only applicable for Service Principal logins.
        type: env_var_name
      azure-tenant:
        default: AZURE_TENANT
        description: |
          Environment variable storing your Azure tenant, necessary if
          `alternate-tenant` is set to true. Only applicable for user logins.
        type: env_var_name
      azure-username:
        default: AZURE_USERNAME
        description: |
          Environment variable storing your Azure username. Only applicable for
          user logins.
        type: env_var_name
      checkout:
        default: true
        description: |
          Boolean for whether or not to checkout as a first step. Default is true.
        type: boolean
      dockerfile:
        default: Dockerfile
        description: Name of dockerfile to use. Defaults to Dockerfile.
        type: string
      executor:
        default: default
        description: executor to use for this job
        type: executor
      extra-build-args:
        default: ''
        description: |
          Extra flags to pass to docker build. For examples, see https://docs.docker.com/engine/reference/commandline/build
        type: string
      login-server-name:
        description: |
          The Azure login server for your ACR instance, e.g. {yourregistryname}.azure.io
        type: string
      path:
        default: .
        description: |
          Path to the directory containing your Dockerfile and build context. Defaults to . (working directory).
        type: string
      registry-name:
        description: The name of your ACR registry
        type: string
      repo:
        description: A URI to an ACR login-server-name/repository
        type: string
      tag:
        default: latest
        description:
          Comma-delimited string containing docker image tags (default
          = latest)
        type: string
      workspace-root:
        default: .
        description: |
          Workspace root path that is either an absolute path or a path relative to the working directory. Defaults to '.' (the working directory)
        type: string
```
It's basically the same step that runs the installation in the `azure-cli` orb but with the version hardcoded in the `apt-get` :)

**Comment by AustinFelipe** [comment]
Awesome! Thanks, man

**Comment by lokst** [comment]
@poloagustin Thank you for reporting the issue! We'll look into how to resolve this and post updates.

**Comment by lokst** [comment]
@poloagustin @AustinFelipe An update on this issue - we are able to resolve the ACR login error by using a Docker credentials store, and should be releasing a fix to the `circleci/azure-acr` orb likely this or next week.

**Comment by lokst** [comment]
This issue should now be resolved in `circleci/azure-acr@0.2.0`
